src/analyses/wrapperFunctionAnalysis.ml

```diff
@@ -0,0 +1,257 @@
+(** An analysis that handles the case when an interesting function is called
+    from a wrapper function all over the code. Currently handles the [malloc]-
+    family of memory allocation functions, as well as [pthread_create] *)
+
+open Prelude.Ana
+open Analyses
+open GobConfig
+open ThreadIdDomain
+module Q = Queries
+
+open Debug
+
+(* Functor argument for creating the chain lattice of unique calls *)
+module type UniqueCountArgs = sig
+  val unique_count : unit -> int
+  val label : string
+  val use_previous_node : bool
+end
+
+(* Functor argument for determining wrapper and wrapped functions *)
+module type WrapperArgs = sig
+  val wrappers : unit -> string list
+  val is_wrapped : LibraryDesc.special -> bool
+end
+
+(* The main analysis, generic to which functions are being wrapped. *)
+module SpecBase (UniqueCountArgs : UniqueCountArgs) (WrapperArgs : WrapperArgs) =
+struct
+  include Analyses.DefaultSpec
+
+  let dbg = WrapperArgs.is_wrapped (LibraryDesc.ThreadCreate { thread = Cil.integer 1; start_routine = Cil.integer 1; arg = Cil.integer 1; })
+  let st name ctx = if dbg then dpf"----------------------------------\n[%s] prev_node=%a => node=%a" name Node.pretty ctx.prev_node Node.pretty ctx.node
+  
+  (* TODO:
+    Does it matter if this is node or prev_node? [malloc] analysis used ctx.node and seemed to care.
+    Thread ID analysis is using ctx.prev_node (which makes more sense, since that's where the thread_create edge is,
+    and would keep two wrapper calls apart if they are e.g. both on edges leading into a join point)
+    https://github.com/goblint/analyzer/commit/77c0423640c50bb82e4290bcc97f33d4082715d0
+      *)
```

Comment:
Intuitively it makes more sense to use `prev_node`. Do things break if one uses that everywhere?

Reply:
The malloc analysis breaks :) You changed it from prev_node in the linked commit, but I can't figure out why the change was made.

Reply:
We determined that the malloc analysis doesn't break (I still had an old test run in mind), so I'll change this to always use prev_node.